libstuff/SQValue.cpp

```diff
@@ -0,0 +1,136 @@
+#include <libstuff/SQValue.h>
+#include <libstuff/sqlite3.h>
+
+// All of our connstructor bodies are emtpy, we just use member initializer lists.
+SQValue::SQValue() : type(SQValue::TYPE::NONE) {}
+SQValue::SQValue(int64_t val) : type(SQValue::TYPE::INTEGER), integer(val) {}
+SQValue::SQValue(double val) : type(SQValue::TYPE::REAL), real(val) {}
+SQValue::SQValue(const char* val) : type(SQValue::TYPE::TEXT), text(val ? val : "") {}
+SQValue::SQValue(const string& val) : type(SQValue::TYPE::TEXT), text(val) {}
+SQValue::SQValue(TYPE t, const string& val) : type(t), text(val) {}
+
+SQValue::operator string() const {
+    switch (type) {
+        case TYPE::TEXT:
+        case TYPE::BLOB:
+            return text;
+        case TYPE::INTEGER:
+            return std::to_string(integer);
+        case TYPE::REAL:
+            // This is the same format that the sqlite3 shell uses internally.
+            char buf[64];
+            sqlite3_snprintf(sizeof(buf), buf, "%!.15g", real);
+            return string(buf);
+        case TYPE::NONE:
+        default:
+            return "";
+    }
+}
+
+string operator+(string lhs, const SQValue& rhs) {
+    lhs += static_cast<string>(rhs);
+    return lhs;
+}
+
+string operator+(const SQValue& lhs, const string& rhs) {
+    return static_cast<string>(lhs) + rhs;
+}
+
+string operator+(const char* lhs, const SQValue& rhs) {
+    return string(lhs) + static_cast<string>(rhs);
+}
+
+string operator+(const SQValue& lhs, const char* rhs) {
+    return static_cast<string>(lhs) + string(rhs);
+}
+
+string operator+(const SQValue& lhs, const SQValue& rhs) {
+    return static_cast<string>(lhs) + static_cast<string>(rhs);
+}
+
+bool operator==(const SQValue& a, const string& b) {
```

Comment:
Any particular reason you switch from lhs/rhc to a/b?

Reply:
I don't have a good answer. Probably I wrote them on different days or something. I can unify them if you like.